src/sidebar/components/sidebar-content.js

```diff
@@ -1,202 +1,165 @@
+import LoginPromptPanel from './login-prompt-panel';
+import SearchStatusBar from './search-status-bar';
+import SelectionTabs from './selection-tabs';
+import SidebarContentError from './sidebar-content-error';
+import ThreadList from './thread-list';
+
+/**
+ * Render the sidebar and its components
+ */
+function SidebarContent({
   frameSync,
-  rootThread,
-  settings,
-  streamer
-) {
-  const self = this;
-
-  this.rootThread = () => rootThread.thread(store.getState());
-
-  function focusAnnotation(annotation) {
-    let highlights = [];
-    if (annotation) {
-      highlights = [annotation.$tag];
-    }
-    frameSync.focusAnnotations(highlights);
-  }
-
-  function scrollToAnnotation(annotation) {
-    if (!annotation) {
-      return;
-    }
-    frameSync.scrollToAnnotation(annotation.$tag);
-  }
-
-  /**
-   * Returns the Annotation object for the first annotation in the
-   * selected annotation set. Note that 'first' refers to the order
-   * of annotations passed to store when selecting annotations,
-   * not the order in which they appear in the document.
-   */
-  function firstSelectedAnnotation() {
-    const selectedAnnotationMap = store.getSelectedAnnotationMap();
-    if (selectedAnnotationMap) {
-      const id = Object.keys(selectedAnnotationMap)[0];
-      return store.getState().annotations.annotations.find(function (annot) {
-        return annot.id === id;
-      });
-    } else {
-      return null;
-    }
-  }
-
-  this.isLoading = () => {
-    if (
-      !store.frames().some(function (frame) {
-        return frame.uri;
-      })
-    ) {
-      // The document's URL isn't known so the document must still be loading.
-      return true;
-    }
-    return store.isFetchingAnnotations();
-  };
-
-  $scope.$on('sidebarOpened', function () {
-    analytics.track(analytics.events.SIDEBAR_OPENED);
-
-    streamer.connect();
-  });
-
-  this.$onInit = () => {
-    // If the user is logged in, we connect nevertheless
-    if (this.auth.status === 'logged-in') {
-      streamer.connect();
-    }
-  };
+  onLogin,
+  onSignUp,
+  loadAnnotationsService,
+  rootThread: rootThreadService,
+  streamer,
+}) {
+  const rootThread = useStore(store =>
+    rootThreadService.thread(store.getState())
+  );
 
-  $scope.$on(events.USER_CHANGED, function () {
-    streamer.reconnect();
+  // store state values
+  const focusedGroupId = useStore(store => store.focusedGroupId());
+  const hasAppliedFilter = useStore(store => store.hasAppliedFilter());
+  const isFocusedMode = useStore(store => store.focusModeEnabled());
+  const annotationsLoading = useStore(store => {
+    return !store.hasFetchedAnnotations() || store.isFetchingAnnotations();
   });
+  const isLoggedIn = useStore(store => store.isLoggedIn());
+  const linkedAnnotationId = useStore(store =>
+    store.directLinkedAnnotationId()
+  );
+  const linkedAnnotation = useStore(store =>
+    store.findAnnotationByID(linkedAnnotationId)
```

Comment:
```suggestion
    linkedAnnotationId ? store.findAnnotationByID(linkedAnnotationId) : null
```

This is inefficient in the case where there is no direct linked annotation, and potentially `findAnnotationByID` could throw an error if the ID were invalid, although the current implementation doesn't.